src/generic_sort.cpp

```diff
@@ -118,10 +120,19 @@ Sort make_generic_sort(SortKind sk, SortVec sorts)
   }
 }
 
+Sort make_generic_sort(Datatype & dt)
```

Comment:
```suggestion
Sort make_generic_sort(Datatype dt)
```

We need to remove the `&` to make `make_generic_sort` match the declaration in the header file. This will get rid of the undefined references in the linking step. With the `&`, we are effectively defining a different overload of the function `make_generic_sort` because `Datatype` and `Datatype&` are two different types. As a result, the linker cannot find the implementation of `Sort make_generic_sort(Datatype dt)`.